src/main/java/bio/terra/cli/serialization/userfacing/UFResourceCloneDetails.java

```diff
@@ -0,0 +1,140 @@
+package bio.terra.cli.serialization.userfacing;
+
+import bio.terra.cli.businessobject.Resource;
+import bio.terra.cli.utils.UserIO;
+import bio.terra.workspace.model.CloneResourceResult;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import bio.terra.workspace.model.ResourceCloneDetails;
+import bio.terra.workspace.model.StewardshipType;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import java.io.PrintStream;
+import java.util.Optional;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.StringEscapeUtils;
+
+@JsonDeserialize(builder = UFResourceCloneDetails.Builder.class)
+public class UFResourceCloneDetails {
+  public final CloneResourceResult result;
+  public final CloningInstructionsEnum cloningInstructions;
+  public final Resource.Type resourceType;
+  public final StewardshipType stewardshipType;
+  public final UUID sourceResourceId;
+  @Nullable public final UUID destinationResourceId;
+  //  public final CloneResourceResult result;
+  @Nullable public final String errorMessage;
+  @Nullable public final String name;
+  @Nullable public final String description;
+
+  public UFResourceCloneDetails(ResourceCloneDetails resourceCloneDetails) {
```

Comment:
I think it would be helpful to also include the google project id for the destination workspace. Also, there are enough properties in the cloned resource that I think we can just include all of them. Since there are already user-facing serialization classes for workspaces and resources, perhaps we can reuse them to avoid redefining very similar sets of properties.

Straw proposal:
```
UFClonedWorkspace {
  UFWorkspace sourceWorkspace;
  UFWorkspace destinationWorkspace;
  boolean isSuccessful;
  List<UFClonedResource> clonedResources;
}

UFClonedResource {
  UFResource sourceResource; // should be cached locally already for the current workspace
  UFResource destinationResource; // alternative UUID destinationResourceId. the full object would give you the cloud info (e.g. bucket name, dataset id), but also probably would require additional calls to wsm
  CloneResourceResult result;
  String errorMessage;
}

in command/workspace/clone
  - Call WSM clone
  - Build the UF objects from the clone response and current workspace
```

Reply:
I'm a little worried about scope creep under this ticket, but this is doable. I need to figure out at which layer these conversions need to go. This would definitely be helpful.

Would the JSON format change as well?